src/main/resources/templates/db_scripts/schema.json

```diff
@@ -103,7 +103,8 @@
     },
     {
       "tableName": "po_line",
-      "fromModuleVersion": 7.0,
+      "customSnippetPath": "data-migration/po_line_table.sql",
+      "fromModuleVersion": "mod-orders-storage-9.0.0",
```

Comment:
I'm not saying this is wrong but I lament the naming of this field - IMO it's confusing at best

Reply:
@craigmcnally  The value is based on [Recommendations for the creation and maintenance of the schema.json file](https://github.com/folio-org/raml-module-builder/blob/master/DB-schema-migration.md#recommendations-for-the-creation-and-maintenance-of-the-schemajson-file). The way I specified version is based on [Example 2](https://github.com/folio-org/raml-module-builder/blob/master/DB-schema-migration.md#example-2).

However the actual version number should be set based on `moduleFrom` and `moduleTo` values which are going to be specified as part of upgrade procedure.
The [TenantAPI#postTenant](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/impl/TenantAPI.java#L243-L247) reads the numbers and then passes them into `SchemaMaker`:
```java
    ...
    previousVersion = entity.getModuleFrom();
    newVersion = entity.getModuleTo();
  }

  SchemaMaker sMaker = new SchemaMaker(tenantId, PostgresClient.getModuleName(), op, previousVersion, newVersion);
  ...
```
The [SchemaMaker#generateDDL](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java#L91) then sets previous version as `version` variable for templates
```java
  templateInput.put("version", pVersion);

  templateInput.put("newVersion", this.newVersion);
```

Please let me know what actually is being specified as part of tenant install API call. In case only numbers are specified, the `fromModuleVersion` has to be changed to `9.0.0` or probably even to `9.0`

Reply:
I just had another thought and looks like okapi API will always include fully qualified modules version for `moduleFrom` and `moduleTo` otherwise it will not be obvious version for which module is specified. This means that RMB's tenant API will work with fully qualified modules versions as well.
According to [“fromModuleVersion” notes](https://github.com/folio-org/raml-module-builder/blob/master/DB-schema-migration.md#frommoduleversion-notes) the RMB uses [ComparableVersion.java](https://github.com/folio-org/raml-module-builder/blob/master/util/src/main/java/org/folio/util/ComparableVersion.java) class to compare versions. This class has `main` method so I tested what is going on when comparing different versions:
- "mod-orders-storage-9.0.0-SNAPSHOT.144" and "8.0"
  > mod-orders-storage-9.0.0-SNAPSHOT.144, == mod-orders-storage-9-snapshot.144-,
     mod-orders-storage-9.0.0-SNAPSHOT.144, < 8.0
- "mod-orders-storage-9.0.0-SNAPSHOT.144" and "1.0"
  > mod-orders-storage-9.0.0-SNAPSHOT.144, == mod-orders-storage-9-snapshot.144-,
     mod-orders-storage-9.0.0-SNAPSHOT.144, < 1.0
- "mod-orders-storage-9.0.0-SNAPSHOT.144" and "mod-orders-storage-9.0.0"
  > mod-orders-storage-9.0.0-SNAPSHOT.144, == mod-orders-storage-9-snapshot.144-,
     mod-orders-storage-9.0.0-SNAPSHOT.144, < mod-orders-storage-9.0.0
- "mod-orders-storage-9.0.0" and "mod-orders-storage-8.0.0"
  > mod-orders-storage-9.0.0, == mod-orders-storage-9-,
     mod-orders-storage-9.0.0, > mod-orders-storage-8.0.0
- "mod-orders-storage-9.0.0-SNAPSHOT.144" and "mod-orders-storage-8.1.0-SNAPSHOT.143"
  > mod-orders-storage-9.0.0-SNAPSHOT.144, == mod-orders-storage-9-snapshot.144-,
     mod-orders-storage-9.0.0-SNAPSHOT.144, > mod-orders-storage-8.1.0-SNAPSHOT.143

Reply:
yeah I think it's the full module id w/ name and version